components/dashboard/src/components/RepositoryFinder.tsx

```diff
@@ -57,46 +64,70 @@ export default function RepositoryFinder(props: RepositoryFinderProps) {
 
     const getElements = useCallback(
         (searchString: string) => {
-            let result: string[];
-            searchString = searchString.trim();
-            if (searchString.length > 1) {
-                result = suggestedRepoURLs.filter((e) => e.toLowerCase().indexOf(searchString.toLowerCase()) !== -1);
-                if (result.length > 200) {
-                    result = result.slice(0, 200);
-                }
-                if (result.length === 0) {
-                    try {
-                        // If the searchString is a URL, and it's not present in the proposed results, "artificially" add it here.
-                        new URL(searchString);
-                        if (!suggestedRepoURLs.includes(searchString)) {
-                            result.push(searchString);
-                        }
-                    } catch {}
-                }
-            } else {
-                result = suggestedRepoURLs.slice(0, 200);
-            }
-
-            return result.map(
-                (e) =>
-                    ({
-                        id: e,
-                        element: (
-                            <div className="flex-col ml-1 mt-1 flex-grow">
-                                <div className="flex">
-                                    <div className="text-gray-700 dark:text-gray-300 font-semibold">
-                                        {stripOffProtocol(e)}
+            const results = filterRepos(searchString, suggestedRepoURLs);
+
+            // With the flag off, we only want to show the suggestedContextURLs
+            if (!includeProjectsOnCreateWorkspace) {
+                return results.map(
+                    (repo) =>
+                        ({
+                            id: repo.url,
+                            element: (
+                                <div className="flex-col ml-1 mt-1 flex-grow">
+                                    <div className="flex">
+                                        <div className="text-gray-700 dark:text-gray-300 font-semibold">
+                                            {stripOffProtocol(repo.url)}
+                                        </div>
+                                        <div className="ml-1 text-gray-400">{}</div>
                                     </div>
-                                    <div className="ml-1 text-gray-400">{}</div>
+                                    <div className="flex text-xs text-gray-400">{}</div>
```

Comment:
**issue**: Minor font size change.

```suggestion
                                    <div className="flex text-sm text-gray-400">{}</div>
```

Reply:
That's the old (currently live) rendering, not the new one. I left that as it was - also, I don't think they render anything (empty `{}` object in the divs).